File: 1.two-sum.cpp
#include<vector>
#include<iostream>
#include<algorithm>
using namespace std;


/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        for(int i=0; i<nums.size(); i++){
            int sum = target - nums[i];
            if(m.find(sum)!=m.end()){
                return {m[sum],i};
            }
            m[nums[i]] = i;
        }
        return {};
    }
};
// @lc code=end

File: 1.two-sum.cs
/*
 * @lc app=leetcode id=1 lang=csharp
 *
 * [1] Two Sum
 */

// @lc code=start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Hashtable m = new Hashtable();
        int n = nums.Length;
        int[] res = new int[2];
        for(int i=0; i<n; i++){
            if(m.ContainsKey(target-nums[i])){
                res[0] = i;
                res[1] =(int) m[target-nums[i]];
                return res;
            }
            
            m[nums[i]]=i;
        }
        return res;
    }
}
// @lc code=end

